diff --git a/port/linux/src/xbox_files.c b/port/linux/src/xbox_files.c
--- a/port/linux/src/xbox_files.c
+++ b/port/linux/src/xbox_files.c
@@ -295,6 +295,8 @@ BOOL WINAPI ReadFileEx(HANDLE handle, LPVOID buffer, DWORD count, LPOVERLAPPED o
 		overlapped->Internal = ERROR_HANDLE_EOF;
 	overlapped->InternalHigh = done;
+	if (overlapped->hEvent)
+		SetEvent(overlapped->hEvent);
 	platform_queue_apc(file_completion_apc, (void *)completion_routine, overlapped, NULL);
 	SetLastError(ERROR_SUCCESS);
 	return TRUE;
@@ -317,6 +319,8 @@ BOOL WINAPI WriteFileEx(HANDLE handle, LPCVOID buffer, DWORD count, LPOVERLAPPED
 		overlapped->Internal = ERROR_SUCCESS : GetLastError();
 	overlapped->InternalHigh = done;
+	if (overlapped->hEvent)
+		SetEvent(overlapped->hEvent);
 	platform_queue_apc(file_completion_apc, (void *)completion_routine, overlapped, NULL);
 	SetLastError(ERROR_SUCCESS);
 	return TRUE;
